web/src/main/java/com/microsoft/applicationinsights/web/spring/internal/InterceptorRegistry.java

```diff
@@ -21,18 +21,27 @@
 
 package com.microsoft.applicationinsights.web.spring.internal;
 
-import org.springframework.context.annotation.Configuration;
-import org.springframework.web.servlet.config.annotation.EnableWebMvc;
-import org.springframework.web.servlet.config.annotation.WebMvcConfigurerAdapter;
 import com.microsoft.applicationinsights.web.spring.RequestNameHandlerInterceptorAdapter;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.WebMvcConfigurationSupport;
 
 /**
  * This class registers the RequestNameHandlerInterceptorAdapter to the interceptors registry.
  * The registration enables the interceptor to extract the http request's controller and action names.
+ *
+ * This class extends {@link WebMvcConfigurationSupport} to add {@link RequestNameHandlerInterceptorAdapter}
+ * instead of overriding @EnableWebMvc annotation. This is necessary to prevent breaking SpringBoot
+ * auto-configuration.
+ *
+ * Recommendation from Spring framework : If WebMvcConfigurer does not expose some more advanced setting
+ * that needs to be configured consider removing the @EnableWebMvc annotation and extending directly from
+ * WebMvcConfigurationSupport or DelegatingWebMvcConfiguration.
+ *
+ * @see <a href="https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/servlet/config/annotation/EnableWebMvc.html">
+ *   https://docs.spring.io/spring/docs/current/javadoc-api/org/springframework/web/servlet/config/annotation/EnableWebMvc.html</a>
  */
-@EnableWebMvc
 @Configuration
-public class InterceptorRegistry extends WebMvcConfigurerAdapter {
+public class InterceptorRegistry extends WebMvcConfigurationSupport {
```

Comment:
@snicoll could you please confirm that the changes to InterceptorRegistry will eliminate the autoconfigure issues that were raised? I followed spring's official documentation for making this change.

Reply:
I don't remember the context, sorry. What autoconfigure isssues?

This code is much better than the previous situation. It should be noted that the doc you've pasted applies primarily to end users and applications. Using `@EnableWebMvc` in a starter is simply not acceptable as it switches off everything Spring Boot does for the user.

Reply:
Extending `WebMvcConfigurationSupport` is like using `@EnableWebMvc`, it will override all Spring Boot auto-configuration for Spring MVC. In this case, implementing `WebMvcConfigurer` or extending `WebMvcConfigurerAdapter` should solve the problem.

Reply:
@snicoll @bclozel thanks this is a very good insight! I kind of felt this when taking a look at the doc but wasn't explicit. I will either implement `WebMvcConfigurer` or `WebMvcConfigurerAdapter`. By the way it already was using `WebMvcConfigurerAdapter`. In this case I can just remove `@EnableWebMvc` annotation and leave this upto end user to put it on one of their configuration classes. Let me know if this makes more sense!

Reply:
I don't think you should mention `@EnableWebMvc` anywhere at all. We'll do something sensible automatically for the user and adding this annotation manually is a corner case.

Reply:
Makes sense. The only question is that what happens when user is using plain Spring MVC instead of springboot ?

Reply:
If they're using plain Spring MVC, none of this apply as the auto-configuration support is in Spring Boot.